provider/core/src/data_provider.rs

```diff
@@ -74,6 +74,20 @@ where
     }
 }
 
+/// A data provider that can determine whether it can load a particular data identifier,
+/// potentially cheaper than actually performing the load.
+pub trait DryDataProvider<M: DataMarker>: DataProvider<M> {
+    /// This method goes through the motions of [`load`], but only returns the metadata.
+    ///
+    /// It must be equivalent to calling `load(req).map(|r| r.metadata)`, but might
+    /// be implemented in a more efficient way.
+    ///
+    /// [`load`]: DataProvider::load
+    fn dry_load(&self, req: DataRequest) -> Result<DataResponseMetadata, DataError> {
+        self.load(req).map(|r| r.metadata)
+    }
```

Comment:
I think this is a good tradeoff between having the method on the `DataProvider` trait with a default impl (in that case you can't get rid of it), and having it on its own trait without a default impl (in that case you always have to implement it yourself). Because it's a single-method trait, you won't accidentally get the default impl.